src/shared/sam/cli/samCliLocator.ts

```diff
@@ -87,16 +91,20 @@ abstract class BaseSamCliLocator {
                 try {
                     const validationResult = await validator.getVersionValidatorResult()
                     if (validationResult.validation === SamCliVersionValidation.Valid) {
+                        BaseSamCliLocator.didFind = true
                         return fullPath
                     }
-                    this.logger.info(`Found invalid SAM executable (${validationResult.validation}): ${fullPath}`)
+                    if (!BaseSamCliLocator.didFind) {
```

Comment:
I would leave this one out of the filtering.

Reply:
reverted